packages/query-core/src/tests/mutations.test.tsx

```diff
@@ -325,7 +325,7 @@ describe('mutations', () => {
       context: undefined,
       variables: undefined,
       data: 'new',
-      error: undefined,
+      error: null,
```

Comment:
I wouldn't expect any changes in tests from a type level change. Why is `error` `null` here instead of `undefined` ?

Reply:
because when we defaulted the type to Error, it's expecting a null value instead of undefined